cmd/mailchain/internal/http/handlers/messages_test.go

```diff
@@ -15,12 +15,229 @@
 package handlers
 
 import (
+	"context"
+	"errors"
+	"github.com/golang/mock/gomock"
+	"github.com/mailchain/mailchain/crypto/cipher"
+	"github.com/mailchain/mailchain/crypto/cipher/ciphertest"
+	"github.com/mailchain/mailchain/internal/clients/etherscan"
+	"github.com/mailchain/mailchain/internal/keystore"
+	"github.com/mailchain/mailchain/internal/keystore/kdf/multi"
+	"github.com/mailchain/mailchain/internal/keystore/keystoretest"
+	"github.com/mailchain/mailchain/internal/mail"
+	"github.com/mailchain/mailchain/internal/mailbox"
+	"github.com/mailchain/mailchain/internal/mailbox/mailboxtest"
+	"github.com/mailchain/mailchain/internal/testutil"
+	"github.com/mailchain/mailchain/stores"
+	"github.com/mailchain/mailchain/stores/storestest"
+	"io/ioutil"
 	"net/http"
+	"net/http/httptest"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
+func Test_GetMessages(t *testing.T) {
+	assert := assert.New(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	type args struct {
+		inbox            stores.State
+		receivers        map[string]mailbox.Receiver
+		ks               keystore.Store
+		deriveKeyOptions multi.OptionsBuilders
+	}
+	tests := []struct {
+		name       string
+		args       args
+		req        *http.Request
+		wantBody   string
+		wantStatus int
+	}{
+		{
+			"err-empty-address",
+			args{},
+			httptest.NewRequest("GET", "/?address=&network=mainnet&protocol=ethereum", nil),
+			"{\"code\":422,\"message\":\"'address' must not be empty\"}\n",
+			http.StatusUnprocessableEntity,
+		},
+		{
+			"err-receiver-not-supported",
+			args{},
+			httptest.NewRequest("GET", "/?address=0x5602ea95540bee46d03ba335eed6f49d117eab95c8ab8b71bae2cdd1e564a761&network=mainnet&protocol=ethereum", nil),
+			"{\"code\":422,\"message\":\"receiver not supported on \\\"ethereum/mainnet\\\"\"}\n",
+			http.StatusUnprocessableEntity,
+		},
+		{
+			"err-receiver-no-configured",
+			args{
+				receivers: map[string]mailbox.Receiver{
+					"ethereum/mainnet": nil,
+				},
+			},
+			httptest.NewRequest("GET", "/?address=0x5602ea95540bee46d03ba335eed6f49d117eab95c8ab8b71bae2cdd1e564a761&network=mainnet&protocol=ethereum", nil),
+			"{\"code\":422,\"message\":\"no receiver configured for \\\"ethereum/mainnet\\\"\"}\n",
+			http.StatusUnprocessableEntity,
+		},
+		{
+			"err-no-private-key-found",
+			args{
+				receivers: map[string]mailbox.Receiver{
+					"ethereum/mainnet": etherscan.APIClient{},
+				},
+				ks: func() keystore.Store {
+					store := keystoretest.NewMockStore(mockCtrl)
+					store.EXPECT().HasAddress([]uint8{86, 2, 234, 149, 84, 11, 238, 70, 208, 59, 163, 53, 238, 214, 244, 157, 17, 126, 171, 149, 200, 171, 139, 113, 186, 226, 205, 209, 229, 100, 167, 97}, "ethereum", "mainnet").Return(false).Times(1)
```

Comment:
`[]byte` are really difficult to read at the best of times. Can you change all the `[]uint8{86, 2, 234, 149, 84, 11, 238, 70, 208, 59, 163, 53, 238, 214, 244, 157, 17, 126, 171, 149, 200, 171, 139, 113, 186, 226, 205, 209, 229, 100, 167, 97}` with `[]byte{0x56, 0x2, 0xea, 0x95, 0x54, 0xb, 0xee, 0x46, 0xd0, 0x3b, 0xa3, 0x35, 0xee, 0xd6, 0xf4, 0x9d, 0x11, 0x7e, 0xab, 0x95, 0xc8, 0xab, 0x8b, 0x71, 0xba, 0xe2, 0xcd, 0xd1, 0xe5, 0x64, 0xa7, 0x61}` It makes it slightly easier to read when comparing it to the hex `address=0x5602ea95540bee46d03ba335eed6f49d117eab95c8ab8b71bae2cdd1e564a761`